posthog/api/event.py

```diff
@@ -306,12 +307,15 @@ def values(self, request: request.Request, **kwargs) -> response.Response:
                     left=ast.Field(chain=["timestamp"]),
                     right=ast.Constant(value=date_to),
                 ),
-                ast.CompareOperation(
-                    op=ast.CompareOperationOp.NotEq,
-                    left=ast.Field(chain=["properties", key]),
-                    right=ast.Constant(value=None),
-                ),
             ]
+            if not is_column:
+                conditions.append(
+                    ast.CompareOperation(
+                        op=ast.CompareOperationOp.NotEq,
+                        left=ast.Field(chain=["properties", key]),
+                        right=ast.Constant(value=None),
+                    )
+                )
```

Comment:
It's probably good for consistency to use the `chain` variable from below here as well.

Reply:
Hm. The left side won't ever be `[key]`, though. I think it's better to have it as-is?

Reply:
It may be worth keeping the non-null check as a default like it used to be (so removing the `if`)... but maybe not. I'm quite indifferent actually, so up to you

Reply:
I'm indifferent, too. Added it with 9bf1d91a0d